modules/tensor_mechanics/include/materials/FiniteStrainHyperElasticViscoPlastic.h

```diff
@@ -0,0 +1,144 @@
+#ifndef FINITESTRAINHYPERELASTICVISCOPLASTIC_H
+#define FINITESTRAINHYPERELASTICVISCOPLASTIC_H
+
+#include "FiniteStrainMaterial.h"
+#include "HyperElasticStress.h"
+#include "FlowRateModel.h"
+
+/**
+ * This class is for Hyperelastic Visco-Plastic material
+ * Multiple flow models are solved simultaneously
+ * The flow models are passed as user objects
+ * Residual equations are set for flow rate equations
+ * Solved using NR with substepping
+ * Stress is calculated using user objects (Present PK2 stress)
+ */
+
+class FiniteStrainHyperElasticViscoPlastic;
+
+template<>
+InputParameters validParams<FiniteStrainHyperElasticViscoPlastic>();
+
+class FiniteStrainHyperElasticViscoPlastic : public FiniteStrainMaterial
+{
+ public:
+  FiniteStrainHyperElasticViscoPlastic (const std::string & name, InputParameters parameters);
+
+ protected:
+  /**
+   * This function computes the Cauchy stress
```

Comment:
Single line doc strings should be inserted using ```///```. Then you can even afford a blank line after each member